### DISCUSSION TITLE
Has the algorithm for generating short names changed between wix3 and wix4?

### DISCUSSION THREAD

**Discussion by appel1** [wixtoolset/issues#7595] [Questions] [answered]
When authoring a patch after updating to wix4 I noticed that all short names for directories and files has changed. Has the algorithm changed, and if so can you configure it to use the old one so I don't have to manually specify all the short names? 

Or is it possible some other change I've done when converting our .wxs files to wix4 has triggered this?

**Comment by robmen** [comment, marked as answer]
When patching, you will want to stick to the same version of the WiX Toolset that you used to build the original package. Patching is super sensitive to all sorts of changes that we don't guarantee to be stable across WiX versions, especially not major WiX versions.

**Reply by appel1** [reply]
Makes sense. We'll stick with wix3 for now then. 

Thanks!

**Reply by appel1** [reply]
I'll have to test, but won't changing only the short name of a file in a component also cause issues for major upgrades if you schedule RemoveExistingProducts after InstallFinalize?

**Reply by robmen** [reply]
Only if you install using short filenames.

**Reply by appel1** [reply]
Hopefully not something clients mess with but I guess it is better to be safe and ensure they do not change. 

Thanks again.

**Reply by robmen** [reply]
Very few modern applications will work if forced to install using short file names because it es extremely rare for developers to limit all of their file names to 8.3. It's just not a realistic issue.

**Reply by appel1** [reply]
Ok, so you get the short names only in that case. Yeah, that'll definitely break things so non issue for us.

Sorry for the noise.
